airbyte-cdk/python/airbyte_cdk/sources/declarative/create_partial.py

```diff
@@ -41,7 +41,7 @@ def newfunc(*fargs, **fkeywords):
 
         # interpolate the parameters
         interpolated_keywords = InterpolatedMapping(fully_created, interpolation).eval(config, **{"options": options})
-        interpolated_keywords = {k: v for k, v in interpolated_keywords.items() if v is not None}
+        interpolated_keywords = {k: v for k, v in interpolated_keywords.items() if v}
```

Comment:
i think we'll be able to get rid of "compile-time" interpolation" once we get the interpolated authenticators in

Reply:
i'm referring to the strings evaluated by jinja, eg "{{ 1 + 1 }}"

they can either be evaluated when parsing the yaml file and instantiating the objects, or at runtime on every request/response.

we generally want the strings to be evaluated at runtime because they can depend on the requests/response/records/etc, but right now we also need to try evaluating them during the parsing step because the authenticators use normal strings, not InterpolatedString.

That won't be the case once https://github.com/airbytehq/airbyte/pull/13993 is merged and we'll be able to delete these lines.

Reply:
nice!